pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -208,6 +216,159 @@ func (ptrs *PipelineTaskRunSpec) convertFrom(ctx context.Context, source v1.Pipe
 	ptrs.ComputeResources = source.ComputeResources
 }
 
+// ConvertTo implements apis.Convertible
+func (prs *PipelineRunStatus) ConvertTo(ctx context.Context, sink *v1.PipelineRunStatus) error {
+	// TODO: handle ducktype conversion using knative func once https://github.com/knative/pkg/issues/2655 is fixed
+	sink.Status.ObservedGeneration = prs.Status.ObservedGeneration
+	sink.Status.Conditions = duckv1.Conditions(prs.Status.Conditions)
+	sink.Status.Annotations = prs.Status.Annotations
+
+	sink.StartTime = prs.StartTime
+	sink.CompletionTime = prs.CompletionTime
+	sink.Results = nil
+	for _, pr := range prs.PipelineResults {
+		new := v1.PipelineRunResult{}
+		pr.convertTo(ctx, &new)
+		sink.Results = append(sink.Results, new)
+	}
+	if prs.PipelineSpec != nil {
+		sink.PipelineSpec = &v1.PipelineSpec{}
+		err := prs.PipelineSpec.ConvertTo(ctx, sink.PipelineSpec)
+		if err != nil {
+			return err
+		}
+	}
+	sink.SkippedTasks = nil
+	for _, st := range prs.SkippedTasks {
+		new := v1.SkippedTask{}
+		st.convertTo(ctx, &new)
+		sink.SkippedTasks = append(sink.SkippedTasks, new)
+	}
+	sink.ChildReferences = nil
+	for _, cr := range prs.ChildReferences {
+		new := v1.ChildStatusReference{}
+		cr.convertTo(ctx, &new)
+		sink.ChildReferences = append(sink.ChildReferences, new)
+	}
+	sink.FinallyStartTime = prs.FinallyStartTime
+	if prs.Provenance != nil {
+		new := v1.Provenance{}
+		prs.Provenance.convertTo(ctx, &new)
+		sink.Provenance = &new
+	}
+	if prs.TaskRuns != nil {
+		sink.ChildReferences = append(sink.ChildReferences, convertTaskRunsToChildReference(ctx, prs.TaskRuns)...)
+	}
+	if prs.Runs != nil {
+		sink.ChildReferences = append(sink.ChildReferences, convertRunsToChildReference(ctx, prs.Runs)...)
+	}
```

Comment:
If `embedded-status` is set to "both", both ChildReferences and TaskRuns/Runs will be populated, resulting in duplicate info in ChildReferences, which isn't what we want. Please also add a test for this

Reply:
Thanks Lee, added.